src/catnip/runtime/memory/manager.rs

```diff
@@ -238,14 +240,16 @@ impl MemoryManager {
         Mbuf::new(mbuf_ptr, self.clone())
     }
 
-    pub fn alloc_sgarray(&self, size: usize) -> dmtr_sgarray_t {
+    pub fn alloc_sgarray(&self, size: usize) -> Result<dmtr_sgarray_t, Fail> {
         assert!(size <= self.inner.config.get_max_body_size());
 
         let sgaseg = if self.inner.config.get_inline_body_size() < size
             && size <= self.inner.config.get_max_body_size()
         {
             let mbuf_ptr = unsafe { rte_pktmbuf_alloc(self.inner.body_pool) };
-            assert!(!mbuf_ptr.is_null());
+            if mbuf_ptr.is_null() {
+                return Err(Fail::new(ENOMEM, "cannot allocate more mbufs"));
```

Comment:
I'd make this text be "mbuf allocation failed".  All we know is that this one attempt to allocate a mbuf failed at this point in time.  The next attempt might succeed.

Reply:
Sure.